src/app/(authenticated)/categories/new/page.tsx

```diff
@@ -1,119 +1,13 @@
-                    marginBottom: '.8rem',
-                },
-            })
-            setLoading(false)
-        }
-    }
-
-    return (
-        <form
-            onSubmit={(e) => {
-                e.preventDefault()
-                createCategory()
-            }}
-            className="mx-auto mt-28 max-w-5xl place-items-stretch lg:grid lg:grid-cols-2"
-        >
-            <div className="m-3 p-1">
-                <label htmlFor="name" className="label">
-                    Nombre
-                </label>
-                <input
-                    id="name"
-                    required
-                    className="input"
-                    type="text"
-                    name="name"
-                    placeholder="Nombre de la categoría"
-                    onChange={(e) =>
-                        setCategory({
-                            ...category,
-                            [e.target.name]: e.target.value,
-                        })
-                    }
-                />
-            </div>
-            <div className="m-3 p-1">
-                <label htmlFor="price" className="label">
-                    Precio hora
-                </label>
-
-                <CurrencyInput
-                    priceSetter={(e: React.ChangeEvent<HTMLInputElement>) =>
-                        setCategory({
-                            ...category,
-                            [e.target.name]: [
-                                {
-                                    from: new Date(),
-                                    price:
-                                        parseLocaleNumber(
-                                            e.target.value,
-                                            'de-DE'
-                                        ) * 10,
-                                    //No le paso la currency porque está por default en ARS.
-                                },
-                            ] as HistoricCategoryPrice[],
-                        })
-                    }
-                />
-            </div>
-
-            <Button
-                intent="secondary"
-                type="submit"
-                className="float-right m-4 lg:col-start-2 lg:col-end-3 lg:place-self-end"
-            >
-                {loading ? (
-                    <span className="loader-primary h-5 w-5"></span>
-                ) : (
-                    'Crear categoría'
-                )}
-            </Button>
-        </form>
-    )
+import { canAccess } from '@utils/scopes'
+import { authOptions } from 'app/api/auth/[...nextauth]/route'
+import CategoryForm from 'modules/categories/category-form'
+import { getServerSession } from 'next-auth'
+import { redirect } from 'next/navigation'
+import React from 'react'
+
+export default async function Page() {
+    const session = await getServerSession(authOptions)
+    if (!session) return
+    if (!canAccess('USERS', session.user.role)) redirect('/protocols')
```

Comment:
Can access "CATEGORIES" not "USERS", if this is a /categories/page

Reply:
Solved